scripts/run_e2e_tests.py

```diff
@@ -409,9 +356,9 @@ def main(args=None):
                 flake_checker.report_pass(parsed_args.suite)
                 break
 
-            # Check whether we should rerun based on this suite's policy
-            # and override instructions from the flake checker server.
-            test_is_flaky, rerun_override = flake_checker.is_test_output_flaky(
+            # Check whether we should rerun based on the instructions from the
+            # flake checker server.
+            _, rerun_override = flake_checker.is_test_output_flaky(
```

Comment:
Should we rename the method to should_test_be_rerun_on_failure (or, if we want to keep the "flaky" concept, is_test_flaky since it refers to the test and not its output)?

Reply:
We're not quite checking whether the test is flaky; instead, we're checking whether the test flaked in this particular run. For example, if a flaky test fails but not because of a flake, this function returns `False` even though the test is flaky. To communicate this nuance, how about `check_if_test_flaked`?

Reply:
I dug into the code a bit and I wonder whether this can be simplified (especially since this is the only place we call that function and now we're ignoring its first return value).

Do we still have the concept of storing whether tests are flaky, especially since we're removing all the rerun policies? Maybe we should just simplify this to something like `should_rerun = flake_checker.should_test_be_rerun(...)`.

Otherwise, the concept is really something like "is_test_flaky_and_failed_in_this_run" which really should be two separate calls.

Reply:
I named the function `check_test_flakiness` because the function does report whether the test flaked (by printing it to the console), not just whether we should rerun it (which is all that gets returned)